File: 645.set-mismatch.py
#
# @lc app=leetcode id=645 lang=python3
#
# [645] Set Mismatch
#

# @lc code=start
from typing import List


class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        hash_set, n, result, total = set(), len(nums), [], 0
        for num in nums:
            if num not in hash_set:
                hash_set.add(num)
                total += num
            else:
                result.append(num)
        result.append(n * (n + 1) // 2 - total)
        return result
# @lc code=end

File: 645.set-mismatch.cpp
/*
 * @lc app=leetcode id=645 lang=cpp
 *
 * [645] Set Mismatch
 */

// @lc code=start
#include <vector>
#include <unordered_set>

class Solution {
public:
    std::vector<int> findErrorNums(std::vector<int>& nums) {
        std::unordered_set<int> hashSet;
        int n = nums.size();
        std::vector<int> result;
        int total = 0;

        for (int num : nums) {
            if (hashSet.find(num) == hashSet.end()) {
                hashSet.insert(num);
                total += num;
            } else {
                result.push_back(num);
            }
        }

        result.push_back(n * (n + 1) / 2 - total);
        return result;
    }
};

// @lc code=end

